pkg/gateway/vald/handler/grpc/handler.go

```diff
@@ -161,15 +161,13 @@ func (s *server) search(ctx context.Context, cfg *payload.Search_Config,
 					return nil
 				}
 				id := dist.GetId()
-				mu.RLock()
+				mu.Lock()
```

Comment:
i'm not sure but how about to use sync.Map as `visited`?

Reply:
we already tried sync.Map implementation, at https://github.com/vdaas/vald/pull/729/files#diff-320187c6fd3d930cdcd5999eb7192ed5R150 and https://github.com/vdaas/vald/pull/729/files#diff-320187c6fd3d930cdcd5999eb7192ed5R172-R175
if we change this logic to sync.Map we have to re-check accuracy.

Reply:
I think if we change mutex to lockfree we should make another PR, this PR is for current bug fix.

Reply:
aha, i see.

Reply:
But I wanted to use sync.Map for this block, I'll make proposal PR for it.